File: app/src/main/java/com/dialogapp/dialog/ui/profilescreen/ProfileFragmentPagerAdapter.java
package com.dialogapp.dialog.ui.profilescreen;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.dialogapp.dialog.ui.profilescreen.favorites.FavoritesFragment;
import com.dialogapp.dialog.ui.profilescreen.following.FollowingFragment;

public class ProfileFragmentPagerAdapter extends FragmentPagerAdapter {

    private final String[] TAB_TITLES = new String[]{"POSTS", "FAVORITES", "FOLLOWING"};
    private String username;

    ProfileFragmentPagerAdapter(FragmentManager fm, String username) {
        super(fm);
        this.username = username;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return ProfileFragment.newInstance(username);
            case 1:
                return new FavoritesFragment();
            case 2:
                return FollowingFragment.newInstance(username);
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return TAB_TITLES.length;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return TAB_TITLES[position];
    }
}

File: app/src/main/java/com/dialogapp/dialog/ui/profile/ProfileFragmentPagerAdapter.kt
package com.dialogapp.dialog.ui.profile

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.dialogapp.dialog.ui.profile.following.FollowingFragment

class ProfileFragmentPagerAdapter(fm: FragmentManager, private val username: String)
    : FragmentPagerAdapter(fm) {

    private val TAB_TITLES = arrayOf("POSTS", "FOLLOWING")

    override fun getCount(): Int {
        return TAB_TITLES.size
    }

    override fun getItem(position: Int): Fragment? {
        return when (position) {
            0 -> ProfilePostsFragment.newInstance(username)
            1 -> FollowingFragment.newInstance(username, false)
            else -> null
        }
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return TAB_TITLES[position]
    }
}